compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeHashVisitor.java

```diff
@@ -414,7 +414,7 @@ public Integer visit(BTupleType type) {
             return 0;
         }
         List<Integer> tupleTypesHashes = getOrderedTypesHashes(type.getTupleTypes());
-        Integer hash = hash(baseHash(type), tupleTypesHashes, visit(type.restType), type.flags);
+        Integer hash = hash(baseHash(type), tupleTypesHashes, visit(type.restType), type.flags, type.tsymbol);
```

Comment:
We don't quite use the symbol in other places, right? How come we don't have the same issue with records, for example? Just wondering if this is the correct place to fix.

Reply:
My thinking is; that it should be updated for other cases as well.  @HindujaB @warunalakshitha 

btw @ushirask whether this is not reproducible for records and other cases?

Reply:
The flags are different for the 2 types in the record case, I'll check and see why it's not the same when in comes to tuples.

Reply:
I am continuing the discussion here which started the below comment.
https://github.com/ballerina-platform/ballerina-lang/pull/36638/files#r940983777

`tsymbol` does not override the hash function. Therefore if we add it like this hash will be the hash of the address value which is going to change from compilation to compilation. We need an invariant value.

Reply:
Check the code in `BIRPackageSymbolEnter` https://github.com/ballerina-platform/ballerina-lang/blob/bbd084cd679015c8aff2ef56d126ded38030a5cd/compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/BIRPackageSymbolEnter.java#L1516

This code is run when reading from BIR instead of Ballerina code. We are not adding a name here and creating a new symbol here.